backend/app/modules/user/antisybil/service/initial.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime
+import time
+from typing import List
+
+from app.extensions import db
+from app.context.manager import Context
+from app.infrastructure import database
+from app.pydantic import Model
+from app.infrastructure.external_api.gc_passport.score import (
+    issue_address_for_scoring,
+    fetch_score,
+    fetch_stamps,
+)
+
+
+class InitialUserAntisybil(Model):
+    def get_antisybil_status(self, _: Context, user_address: str) -> (int, datetime):
+        score = database.user_antisybil.get_score_by_address(user_address)
+        return score.score, score.expires_at
+
+    def fetch_antisybil_status(
+        self, _: Context, user_address: str
+    ) -> (str, datetime, any):
+        score = issue_address_for_scoring(user_address)
+
+        while score["status"] != "DONE":
+            print("Waiting for score for user {user_address}")
+            time.sleep(3)
+            score = fetch_score(user_address)
+
+        all_stamps = fetch_stamps(user_address)["items"]
+        cutoff = datetime.now()
+        valid_stamps = _filter_older(cutoff, all_stamps)
+        expires_at = datetime.now()
+        if len(valid_stamps) != 0:
+            expires_at = _parse_expirationDate(
+                min([stamp["credential"]["expirationDate"] for stamp in valid_stamps])
+            )
+        return score["score"], expires_at, all_stamps
+
+    def update_antisybil_status(
+        self,
+        context: Context,
+        user_address: str,
+        score: str,
+        expires_at: datetime,
+        stamps,
+    ):
+        database.user_antisybil.add_score(user_address, score, expires_at, stamps)
+        db.session.commit()
+
+
+def _parse_expirationDate(timestamp_str):
+    # GP API returns expirationDate in both formats
+    formats = ["%Y-%m-%dT%H:%M:%S.%fZ", "%Y-%m-%dT%H:%M:%SZ"]
```

Comment:
you can rename this variable and then remove the comment from 54. line

Reply:
:+1: